source/fab/tools/tool_repository.py

```diff
@@ -176,7 +184,12 @@ def get_default(self, category: Category,
 
         # If not a compiler or linker, return the first tool
         if not category.is_compiler and category != Category.LINKER:
-            return self[category][0]
+            for tool in self[category]:
+                if tool.is_available:
+                    return tool
+            tool_names = ",".join(i.name for i in self[category])
+            raise RuntimeError(f"Can't find available '{category}' tool. "
+                               f"Tools are '{tool_names}'.")
```

Comment:
This is new behaviour extraneous to  the description. Please add it to the description.

Reply:
Exception description added, also added a sentence or two to the docs.